numpy/core/src/multiarray/getset.c

```diff
@@ -472,25 +472,19 @@ array_descr_set(PyArrayObject *self, PyObject *arg)
         Py_DECREF(safe);
     }
 
-    if (newtype->elsize == 0) {
-        /* Allow a void view */
-        if (newtype->type_num == NPY_VOID) {
-            PyArray_DESCR_REPLACE(newtype);
-            if (newtype == NULL) {
-                return -1;
-            }
-            newtype->elsize = PyArray_DESCR(self)->elsize;
-        }
-        /* But no other flexible types */
-        else {
-            PyErr_SetString(PyExc_TypeError,
-                    "data-type must not be 0-sized");
-            Py_DECREF(newtype);
+    /*
+     * Treat V0 as resizable void - unless the destination is already V0, then
+     * don't allow np.void to be duplicated
+     */
+    if (newtype->type_num == NPY_VOID && newtype->elsize == 0
+                                      && PyArray_DESCR(self)->elsize != 0) {
```

Comment:
That doesn't match with what I remember seeing, eg in `mapping.c`. Operators should be at the start of the line, so that they're not missed when reading through.

I can wrap after `NPY_VOID` if you'd like, but I'm not sure what level to indent to as a result